### PR TITLE
Fix for 3.18

### LINKED ISSUE #66: 0.0.6 not building against NXRM 3.18.1
Thanks for creating an issue! Please fill out this form so we can be
sure to have all the information we need, and to minimize back and forth.

* **What are you trying to do?**
Trying to build the project against NXRM 3.18.1

* **What feature or behavior is this required for?**
Builds correctly and cleanly

* **How could we solve this issue? (Not knowing is okay!)**

* **Anything else?**
Build output:
```[ERROR] Found 3 errors and 0 warnings.
[INFO] 4 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:10 min
[INFO] Finished at: 2019-09-17T16:05:59Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project nexus-repository-conan: Compilation failure: Compilation failure: 
[ERROR] /nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java:[31,21] 1. ERROR in /nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java (at line 31)
[ERROR]         public List<String> computeComponentPath(final Asset asset, final Component component) {
[ERROR]                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The method computeComponentPath(Asset, Component) of type ConanBrowseNodeGenerator must override or implement a supertype method
[ERROR] 
[ERROR] /nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java:[48,21] 2. ERROR in /nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java (at line 48)
[ERROR]         public List<String> computeAssetPath(final Asset asset, final Component component) {
[ERROR]                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The method computeAssetPath(Asset, Component) of type ConanBrowseNodeGenerator must override or implement a supertype method
[ERROR] 
[ERROR] /nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java:[56,14] 3. ERROR in /nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java (at line 56)
[ERROR]         return super.computeAssetPath(asset, component);
[ERROR]                      ^^^^^^^^^^^^^^^^
[ERROR] The method computeAssetPath(Asset, Component) is undefined for the type ComponentPathBrowseNodeGenerator
[ERROR] 
[ERROR] Found 3 errors and 0 warnings.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

### PR DESCRIPTION
**Pull request by areian**
Updated Conan plugin to compile against Nexus v3.18

This pull request makes the following changes:
* Change compute*Path returning List<String> to compute*Paths returning List<BrowsePaths>
* A bit of clean up of variables and helper functions, brought on by the change above
* Updated POM and Dockerfile to compile against Nexus v3.18.1

It relates to the following issue:
* Fixes #66

### CODE REVIEW

File: src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java [hunk 1]
```diff
@@ -28,33 +27,41 @@ public ConanBrowseNodeGenerator() {
...
+    path.add(component.version());
+    return path;
+  }
+
+  private List<String> createAssetPathList(final Asset asset, final Component component) {
+    List<String> path = createComponentPathList(asset, component);
+
+    String[] assetSegments = asset.name().split("/");
+    if(asset.name().contains("packages")) {
+      path.add(assetSegments[assetSegments.length-4]);
```
Comment (dbradicich) [inline src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java]: may want to validate the number of segments to be safe

File: src/test/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGeneratorTest.java [hunk 2]
```diff
@@ -45,9 +47,25 @@ public void canComputeNonPackagePath() {
...
 
-    List<String> assetPath = underTest.computeAssetPath(asset, component);
+    List<BrowsePaths> assetPath = underTest.computeAssetPaths(asset, component);
 
-    assertThat(assetPath, contains("vthiery", "jsonformoderncpp", "2.1.1", "stable", "download_urls"));
+    assertPaths(assetPath, asList("vthiery", "jsonformoderncpp", "2.1.1", "stable", "download_urls"));
+  }
+
+  private void assertPaths(List<BrowsePaths> paths, List<String> expectedPaths) {
+    assertThat(expectedPaths.size(), is(expectedPaths.size()));
```
Comment (dbradicich) [inline src/test/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGeneratorTest.java]: this assert doesn't seem particularly useful ;)

Comment (areian) [reply src/test/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGeneratorTest.java]: Woops! Yeah, I'll get that fixed 😁

File: src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java (hunk 1 above)
Comment (DarthHater) [reply src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java]: Pretty sure this is resolved now by the:

```
if(assetSegments.length < 2) {
      return path;
    }
```

File: src/test/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGeneratorTest.java (hunk 2 above)
Comment (DarthHater) [reply src/test/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGeneratorTest.java]: Resolved in: db950ba

File: src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java (hunk 1 above)
Comment (DarthHater) [reply src/main/java/org/sonatype/repository/conan/internal/ui/ConanBrowseNodeGenerator.java]: In: a37c0d8

**Pull request merged**